extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -1116,9 +1207,39 @@ class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
 }
 
 /**
- * DSL function for [GeoJsonSource].
+ * DSL function for [GeoJsonSource] performing parsing using background thread.
+ * Immediately returns [GeoJsonSource] with no data set and starts preparing actual data
+ * using a worker thread.
+ *
+ * If using runtime styling:
+ *
+ * loadStyle(style(Style.DARK) {
+ *   +geoJsonSource(id) {
+ *    featureCollection(collection)
+ *   }
+ *   ...
+ * }
+ *
+ * compositing style will be performed correctly under the hood.
+ * [Style.OnStyleLoaded] will be emitted in correct moment of time when all sources are parsed.
+ */
+fun geoJsonSource(
+  id: String,
+  block: GeoJsonSource.Builder.() -> Unit
+): GeoJsonSource = GeoJsonSource.Builder(id) {}.apply(block).build()
+
+/**
+ * DSL function for [GeoJsonSource] performing parsing using background thread.
+ * Immediately returns [GeoJsonSource] with no data set,
+ * fully parsed [GeoJsonSource] is returned in [onGeoJsonParsed] callback.
+ *
+ * Using this method means that it is user's responsibility to proceed with adding layers or
+ * other style objects in [onGeoJsonParsed] callback.
  */
-fun geoJsonSource(id: String, block: GeoJsonSource.Builder.() -> Unit): GeoJsonSource =
-  GeoJsonSource.Builder(id).apply(block).build()
+fun geoJsonSource(
+  id: String,
+  block: GeoJsonSource.Builder.() -> Unit,
```

Comment:
`block` naming is well chosen but I'm wondering if calling it async would be even more clear:
```
    geoJsonSource(
      id = "source5",
      async = {
        featureCollection(routePoints)
      },
      onGeoJsonParsed = {
        style.addSource(it)
        style.addLayer(
          lineLayer("layer5", "source5") {
            lineColor("green")
            lineOffset(20.0)
          }
        )
      }
    )
```

Reply:
actually the block itself isn't executed on asynchronously, only the result coming out of it is asynchronously parsed. So  `onAsyncParse` naming would make more sense?

Reply:
Well, all the properties are applied synchronously except of setting `data`. Data could be represented by `feature`, `featureCollection` or `geometry` while all of them are actually representing `GeoJson`. Thus I think that `block` may be renamed to `builderBlock` and `onGeoJsonParsed` sounds OK. Using word `async` here is pretty pointless because docs are clearly stating that this function is async in any case.

Reply:
shall we rename async to for example `configs =` ?

Reply:
yes, will do for overloaded function where both lambda blocks are used for geojson, thanks!